src/core/Logger.cpp

```diff
@@ -130,6 +127,14 @@ LockedLogOutput Logger::log(LogSeverity sev)
 // set_color_enabled turns ANSI colorized output on or off.
 void Logger::set_color_enabled(bool enabled)
 {
+#ifdef _WIN32
+	// Enable ANSI colors on newer versions of Windows
+	HANDLE hOut = GetStdHandle(STD_OUTPUT_HANDLE);
+	DWORD mode = 0;
+	GetConsoleMode(hOut, &mode);
+	mode |= 0x0004; // ENABLE_VIRTUAL_TERMINAL_PROCESSING
```

Comment:
Versions of Windows before 10 do not implement this attribute; it's pretty much only there to support the Windows Subsystem for Linux. Likewise, not all SDK versions define this value (which is probably the reason for the magic number instead of using the symbol).

The recommended way to achieve coloring in Windows is via the `SetConsoleTextAttribute` API. I believe this works on all versions of Windows NT.